### ISSUE TITLE
Understanding the output on terminal while training

### ISSUE THREAD

**Issue by shahaamirbader** [DarkDawn233/SeCA#1]
Hi,

I am in the process of running the training code. While training this is what I see at the output terminal. Since I am really new to SMAC, I have a few questions:

1. What is the best way to understand what these outputs values means? 
2. What should be target values of these outputs during training in order to know if the training is going well or not.
3. Which of these results should match the results shared in paper? Or are the results in paper different from the training results?
4. What does the t_env and episode represents? Why is t_env a total of 656000000. Will it be different for a different map?

Any help in understanding these outputs will be appreciated. 

![Screenshot 2023-10-02 at 9 54 19 PM](https://github.com/DarkDawn233/SeCA/assets/78397774/f22b29ae-c68f-45ca-a60f-f3473ff6d59a)

5. Is it normal for the Starcraft II to take a long time to connect to a game. I repeatedly see the CloseHandler being disconnected after launching the game as follows 

```Game has started.
Using default stable ids, none found at: /homelocal/3rd Year/DT_RL/SeCA/3rdparty/StarCraftII/stableid.json
Successfully loaded stable ids: GameData\stableid.json
Sending ResponseJoinGame
DataHandler: unable to parse websocket frame.
CloseHandler: 127.0.0.1:49004 disconnected
```

**Comment by DarkDawn233** [comment]
这是来自QQ邮箱的自动回复邮件。我已收到邮件，一定尽快查看。

**Comment by DarkDawn233** [comment]
1. The name of the variable indicates the meaning. Variables beginning with `test_` indicate test metrics and the rest are training metrics. We are mainly concerned with the win rate, i.e. `test_battle_win_mean`. The calculation of these metrics is detailed in the source code.
2. The experimental results in the paper are `test_battle_win_mean`, which is the main metric.
3. See 2. 
4. `t_env` represents the number of environment execution steps, i.e., transition. `episode` contains multiple `t_env`s, representing the number of game rounds. Different `t_env` is because different maps have different difficulties and require different convergence times. 
5. It's normal. See details about SMAC environment at: [https://github.com/oxwhirl/smac](https://github.com/oxwhirl/smac).
More details about training framework are available at: [https://github.com/oxwhirl/pymarl](https://github.com/oxwhirl/pymarl)

**Comment by shahaamirbader** [comment]
Thanks. I have another question. I am not able to see the Starcraft game while training. What changes do I need to make to be able to see the game during training or testing?

**Comment by DarkDawn233** [comment]
This is an auto-reply email.I have received the email and will check it as soon as possible.

**Comment by DarkDawn233** [comment]
Please see README [Saving Starcraft II Replay](https://github.com/DarkDawn233/SeCA#saving-starcraft-ii-replay), or PyMARL's README [Watching StarCraft II replays](https://github.com/oxwhirl/pymarl#watching-starcraft-ii-replays).

**Comment by shahaamirbader** [comment]
Thanks but I am confused in the Readme, at the part where it says "and use the episode_runner.". How can I make sure that I am using the episode_runner? Can you kindly share what do I need to change to ensure that I am using episode_runner. Many thanks

**Comment by DarkDawn233** [comment]
Change `runner` in `src/config/algs/seca.yaml`. Please read source code first.
